src/testCommon/java/uk/gov/hmcts/darts/test/common/MemoryLogAppender.java

```diff
@@ -33,6 +33,16 @@ public List<ILoggingEvent> searchLogs(String string, Level level) {
             .collect(Collectors.toList());
     }
 
+    public List<ILoggingEvent> searchLogs(String string, String causeMessage, Level level) {
+        synchronized (GENERAL_LOGS) {
```

Comment:
I notice the other methods in this class don't invoke `synchronized` - would it be better to use a thread safe implementation of the `GENERAL_LOGS` list, which should solve thread safety for all the usages?

Reply:
it would. Done